riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -50,46 +47,61 @@ class DeployFailureNotifications(config: Config,
+       for {
+        yaml <- targetResolver.fetchYaml(parameters.build)
+        deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
+      } yield {
+        TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
+          val maybeAccountId = uuid.flatMap(uuid => getAwsAccountIdTarget(target, parameters, uuid)).toList
+          List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+        }
       }
     }
+    attemptToDeriveTargets match {
+      case Some(Right(targets)) => targets
+      case _ => riffRaffTargets
+    }
+  }
+
+  def notifyViaAnghammarad(notificationContents: NotificationContents, targets: List[Target]) = {
+    log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
+    Anghammarad.notify(
+      subject = notificationContents.subject,
+      message = notificationContents.message,
+      sourceSystem = "riff-raff",
+      channel = All,
+      target = targets,
+      actions = notificationContents.actions,
+      topicArn = anghammaradTopicARN,
+      client = snsClient
+    ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad)", ex) }
+  }
 
-    deriveAnghammaradTargets match {
-      case Right(targets) =>
-        log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
-        val failureMessage = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed."
-        Anghammarad.notify(
-          subject = s"${parameters.deployer.name} failed",
-          message = failureMessage,
-          sourceSystem = "riff-raff",
-          channel = All,
-          target = targets,
-          // TODO: Figure out the correct action for scheduled deploys that don't start
-          actions = uuid.map(id => List(Action("View failed deploy", url(id)))).getOrElse(Nil),
-          topicArn = anghammaradTopicARN,
-          client = snsClient
-        ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad) for $uuid", ex) }
-      case Left(_) =>
-        log.error(s"Failed to derive targets required to notify about failed scheduled deploy: $uuid")
+  def deployUnstartedNotification(error: Error): Unit = {
+    error.scheduledDeployError.map { failedToStartReason =>
+      val contentsWithTargets = deployUnstartedNotificationContents(failedToStartReason, prefix, getTargets, riffRaffTargets)
+      notifyViaAnghammarad(contentsWithTargets.notificationContents, contentsWithTargets.targets)
+    }.getOrElse {
+      log.warn("Scheduled deploy failed to start but notification was not sent...")
     }
   }
 
+  def deployFailedNotification(uuid: UUID, parameters: DeployParameters, targets: List[Target]): Unit = {
+    val notificationContents = deployFailedNotificationContents(uuid, parameters, prefix)
```

Comment:
Personally I prefer refering explicitly to methods in an external object (i.e. qualified with the object name) rather than importing the methods. Makes it easier to comprehend.